UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerManagePage.cs

```diff
@@ -208,13 +208,19 @@ private void SendTeleportAllPlayersToPlayer()
 
 		public void GiveItemToPlayerButton()
 		{
-			var player = PlayerList.Instance.GetPlayerByID(PlayerEntry.PlayerData.uid);
-			if (player == null)
+			adminTools.giveItemPage.selectedPlayer = null;
+			var players = FindObjectsOfType<PlayerScript>(); //since this is client sided it's fiinnnneee
+			foreach (var possiblePlayer in players)
 			{
-				Logger.LogWarning("Unable to find player to give item to! Are you sure that they're still on the list?");
+				if(possiblePlayer.connectedPlayer.UserId != PlayerEntry.PlayerData.uid) continue;
```

Comment:
connectedPlayer could be potentially null?

Reply:
might be funny scenario where, it's a disconnected player

Reply:
I dont think an admin would want to give an offline player an item.. unless you mean that they disconnect and somehow the script doesnt get updated when they re-connect

Reply:
well it's looping through all the players isn't it so that would include off-line players, to find the player it's looking for?

Reply:
it's looping through all player objects in the scene, I see no reason for why the player script would turn null if someone disconnects or anything because thats something that the game already spawns them with, right?

Reply:
wait does the `ConnectedPlayer` thing automatically yeets itself out of the game after someone disconnects?

Reply:
not too sure, I suspect so